### ISSUE TITLE
Ruby 2.2.0 crash with Style/SpaceAroundBlockParameters and stabby lambdas

### ISSUE THREAD

**Issue by ggilder** [bbatsov/rubocop#1675]
The following code sample:

```ruby
->() {
  "hi"
}
```

causes this Rubocop crash on Ruby 2.2.0:

```
Inspecting 1 file
Scanning /Users/gabriel/Development/web/foo.rb
For /Users/gabriel/Development/web: configuration from /Users/gabriel/Development/web/.rubocop.yml
Inheriting configuration from /Users/gabriel/Development/web/.rubocop_todo.yml
Default configuration from /Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/config/default.yml
Inheriting configuration from /Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/config/enabled.yml
Inheriting configuration from /Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/config/disabled.yml
An error occurred while Style/SpaceAroundBlockParameters cop was inspecting /Users/gabriel/Development/web/foo.rb.
undefined method `loc' for nil:NilClass
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/cop/style/space_around_block_parameters.rb:40:in `check_inside_pipes'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/cop/style/space_around_block_parameters.rb:23:in `on_block'
(eval):5:in `block (2 levels) in on_block'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/cop/commissioner.rb:92:in `with_cop_error_handling'
(eval):4:in `block in on_block'
(eval):2:in `each'
(eval):2:in `on_block'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/ast-2.0.0/lib/ast/processor.rb:239:in `process'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/cop/commissioner.rb:52:in `investigate'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/cop/team.rb:33:in `inspect_file'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:126:in `inspect_file'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:97:in `block in do_inspection_loop'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:91:in `loop'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:91:in `do_inspection_loop'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:69:in `process_file'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:38:in `block in run'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:36:in `each'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/runner.rb:36:in `run'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/lib/rubocop/cli.rb:24:in `run'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/bin/rubocop:13:in `block in <top (required)>'
/Users/gabriel/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/Users/gabriel/.rvm/gems/ruby-2.2.0/gems/rubocop-0.29.0/bin/rubocop:12:in `<top (required)>'
/Users/gabriel/.rvm/gems/ruby-2.2.0/bin/rubocop:23:in `load'
/Users/gabriel/.rvm/gems/ruby-2.2.0/bin/rubocop:23:in `<main>'
/Users/gabriel/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
/Users/gabriel/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/SpaceAroundBlockParameters cop was inspecting /Users/gabriel/Development/web/foo.rb.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.29.0 (using Parser 2.2.0.2, running on ruby 2.2.0 x86_64-darwin13)
Finished in 0.03515508503187448 seconds
```

cc @tamird

**Comment by bbatsov** [comment]
Is this broken on 0.29.1 as well?

**Comment by ggilder** [comment]
It seems to be fixed on 0.29.1! Thanks!
